### PR TITLE
Implement deserialization failure handler

### PR DESCRIPTION
**Pull request by cescoffier**

### CODE REVIEW

File: documentation/src/main/doc/modules/connectors/partials/META-INF/connector/smallrye-kafka-incoming.adoc [hunk 1]
```diff
@@ -109,7 +109,15 @@ Type: _int_ | false | `1`
 
 Type: _boolean_ | false | `true`
 
-| *consumer-rebalance-listener.name* | The name set in `javax.inject.Named` of a bean that implements `io.smallrye.reactive.messaging.kafka.KafkaConsumerRebalanceListener`. If set the listener will be applied to the consumer.
+| *consumer-rebalance-listener.name* | The name set in `javax.inject.Named` of a bean that implements `io.smallrye.reactive.messaging.kafka.KafkaConsumerRebalanceListener`. If set, this rebalance listener is be applied to the consumer.
```
Comment (kenfinnigan) [inline documentation/src/main/doc/modules/connectors/partials/META-INF/connector/smallrye-kafka-incoming.adoc:112]: "is be applied" -> could you remove the "be"?

File: documentation/src/main/doc/modules/kafka/pages/inbound.adoc [hunk 2]
```diff
@@ -137,6 +137,40 @@ The record written on the dead letter queue contains a set of additional headers
...
+            Exception exception, Headers headers) {
+        return fallback;
+    }
+}
+----
+
+The bean must be exposed with the `@Named` qualifier specifying the name of the bean.
+Then, in the connector configuration, specify the following attribute:
+
+* `mp.messaging.incoming.$channel.value-deserialization-failure-handler`: name of the bean handling deserialization failures happening for the record's value,
```
Comment (kenfinnigan) [inline documentation/src/main/doc/modules/kafka/pages/inbound.adoc:167]: Not huge, but maybe switch the ordering of the config keys to have the "key" first as it's mentioned first in other situations

**Review by kenfinnigan — APPROVED** [review]
Couple of minor wording changes

File: documentation/src/main/doc/modules/kafka/pages/inbound.adoc (hunk 2 above)
Comment (cescoffier) [reply documentation/src/main/doc/modules/kafka/pages/inbound.adoc:167]: Good point!

**Pull request merged**